lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,24 @@
-
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
-end
\ No newline at end of file
+  if string.empty?
+    return true
+  end
+  #split the string
+  string_array = string.split("")
+
+  #put the sting into a hash
+  hash = Hash.new(0)
+  string_array.each do |letter|
+    hash[letter] += 1
+  end
+  
+
+  #key must have an even value number or one key must only have 1 as value to be considered a palindrome
+  values = hash.values
+  number_1_value = values.count(1)
```

Comment:
This assumes that the letter that occurs an odd number of times only occurs once.  What about `heeeh`?  Or `lool`?

You would do better to count the number of letters that occur an odd number of times.